apps/freenet-ping/app/src/main.rs

```diff
@@ -0,0 +1,134 @@
+use std::time::Duration;
+
+use chrono::Utc;
+use clap::Parser;
+use freenet_ping_types::Ping;
+use freenet_stdlib::{
+    client_api::{ClientRequest, ContractRequest, ContractResponse, HostResponse, WebApi},
+    prelude::{
+        ContractContainer, Parameters, RelatedContracts, StateDelta, UpdateData, WrappedState,
+    },
+};
+use names::Generator;
+
+#[derive(clap::Parser)]
+struct Args {
+    #[clap(long, default_value = "localhost:50509")]
+    host: String,
+    #[clap(long, default_value = "info")]
+    log_level: tracing::level_filters::LevelFilter,
+    #[clap(short, long, default_value = "freenet-ping")]
+    parameters: String,
+}
+
+#[tokio::main(flavor = "multi_thread")]
+async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    let args = Args::parse();
+    tracing_subscriber::fmt()
+        .with_ansi(true)
+        .with_level(true)
+        .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+        .with_max_level(args.log_level)
+        .with_line_number(true)
+        .init();
+
+    const PING_CODE: &[u8] =
+        include_bytes!("../../contracts/ping/build/freenet/freenet_ping_contract");
+
+    // create a websocket connection to host.
+    let uri = format!(
+        "ws://{}/contract/command?encodingProtocol=native",
+        args.host
+    );
+    let (stream, _resp) = tokio_tungstenite::connect_async(&uri).await.map_err(|e| {
+        tracing::error!(err=%e);
+        e
+    })?;
+    let mut client = WebApi::start(stream);
+    // put contract first
+    let params = Parameters::from(args.parameters.into_bytes());
```

Comment:
So after thinking about this more, in the script we could do the following:

We take as parameters in the script: 
- ping frequency (in seconds)
- Record age (how long we keep records)
And then use those as contract params (create a new struct for contract config), this will allow us to easily do tests and parametrize them.

Upon script start, we try to get the existing contract with those parameters, if it doesn't exist, put it, then start the update loop.